### PR TITLE
Document data docker schema update mode

### PR DESCRIPTION
**Pull request by hqpho**
This mode is added by https://github.com/datacommonsorg/website/pull/4686. A subsequent PR to mixer will link the new docsite page directly from the schema check error message: https://github.com/datacommonsorg/mixer/pull/1440

### CODE REVIEW

File: custom_dc/database_update.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+nav_order: 9
+parent: Build your own Data Commons
+---
+
+{:.no_toc}
+# Update your database schema
+
+While starting Data Commons services, you may see an error that starts with `SQL schema check failed`. This means your database schema must be updated for compatibility with the latest Data Commons services.
+
+You can update your database by running a data management job with the environment variable `SCHEMA_UPDATE_ONLY` set to `true`. This will alter your database without modifying already-imported data.
```
Comment (keyurva) [inline custom_dc/database_update.md:13]: May need to be updated based on what we decide in https://github.com/datacommonsorg/website/pull/4686/files#r1815646141

**Review by keyurva — APPROVED** [review]
LGTM modulo the decision on the env var.

File: custom_dc/database_update.md (hunk 1 above)
Comment (hqpho) [reply custom_dc/database_update.md]: Updated.

**Comment by hqpho** [comment]
Will wait to commit this until the mode is available in the stable data container image.

**Review by kmoscoe — COMMENTED** [review]
I wouldn't actually include this as a separate page. It's too minor a detail. I would instead just put the text in the relevant existing sections. Can I send you an alternative PR as a proof of concept?

**Comment by hqpho** [comment]
> I wouldn't actually include this as a separate page. It's too minor a detail. I would instead just put the text in the relevant existing sections. Can I send you an alternative PR as a proof of concept?

My thought was to have it on one page for ease of linking from an error message! Is that alright? Maybe we can remove this page from the sidenav if we don't want to feel like we're cluttering the docsite?

**Comment by kmoscoe** [comment]
You could still link to a subsection using an anchor.

**Comment by kmoscoe** [comment]
Another thing I don't really like is repeating in entirety the startup procedures. It makes more sense to integrate this info into the existing procedures. Let me send you a PR so you can see what I mean.

**Comment by hqpho** [comment]
Kara, thanks for putting together an [alternate approach](https://github.com/datacommonsorg/docsite/pull/529)! I understand the desire not to duplicate startup commands across multiple pages. That said, I worry that the other approach:
- Adds cognitive overhead for people running a typical data load command, trying to copy/paste from a doc page and having to learn about a new param that is almost never relevant to them
- Doesn't explain the mechanism by which the startup time is minimized, so it might come as a surprise to people what exactly the mode does (or rather doesn't) do

Another argument I'd make in favor of having schema update mode be its own page is that we may expand the mode variable in the future to support more different modes, in which case we can easily revise and extend that page.

@keyurva Do you want to weigh in as a decision tiebreaker here?

**Comment by kmoscoe** [comment]
On Mon, Oct 28, 2024 at 3:35 PM Hannah Pho ***@***.***> wrote:

> Kara, thanks for putting together an alternate approach
> <https://github.com/datacommonsorg/docsite/pull/529>! I understand the
> desire not to duplicate startup commands across multiple pages. That said,
> I worry that the other approach:
>
>    - Adds cognitive overhead for people running a typical data load
>    command, trying to copy/paste from a doc page and having to learn about a
>    new param that is almost never relevant to them
>
> OK, another possibility is to have a subheading about running in schema
update mode.

>
>    - Doesn't explain the mechanism by which the startup time is
>    minimized, so it might come as a surprise to people what exactly the mode
>    does (or rather doesn't) do
>
> But that wasn't really given in your PR either. I can easily add some more
info if you give it to me.

>
>
> Another argument I'd make in favor of having schema update mode be its own
> page is that we may expand the mode variable in the future to support more
> different modes, in which case we can easily revise and extend that page.
>
I don't like the idea of having pages determined by some random feature.
They should be determined by the general stage in the workflow, which is
the overall structure I've set up and which is linked from the landing
page. Let's please not introduce additional pages for every new feature we
add.

**Comment by keyurva** [comment]
Is there a middle ground where we don't have a separate page for it but are able to make it as self-sufficient as possible?

The typical workflow is likely going to be what Hannah described: a user starts the service, notices the schema failure with a link to the relevant section / page in the docsite and is primarily interested in resolving this failure in the quickest possible manner.

**Comment by kmoscoe** [comment]
Well, the quickest possible way would just say in the error message:
"Restart the data management job, optionally adding the -e
DATA_RUN_MODE=schemaupdate option for faster performance, and rerun the
services job" or something like that.

The middle ground would be a self-contained topic within the existing
pages. Let me prepare a PR for you that would show that as a PoC, OK?

**Comment by kmoscoe** [comment]
Hey both -- I sent you PR 530
<https://github.com/datacommonsorg/docsite/pull/530> to try to make the
text more standalone. Please review; thanks!

**Comment by hqpho** [comment]
Subsumed by https://github.com/datacommonsorg/docsite/pull/530

**Pull request closed without merging**